src/Drivers/LaravelHttpServer.php

```diff
@@ -222,22 +222,35 @@ private function handleRequest(AmpRequest $request): Response
             Execution::instance()->tick();
         }
 
-        $filepath = public_path($path = $request->getUri()->getPath());
-
-        if (file_exists($filepath) && ! is_dir($filepath)) {
+        $uri = $request->getUri();
+        $path = $uri->getPath() ?: '/';
+        $query = $uri->getQuery() ?? '';
+        $fullPath = $path . ($query !== '' ? '?' . $query : '');
+        $absoluteUrl = rtrim($this->url(), '/') . $fullPath;
+
+        $filepath = public_path($path);
+        if (file_exists($filepath) && !is_dir($filepath)) {
             return $this->asset($filepath);
         }
 
         $kernel = app()->make(HttpKernel::class);
 
+        $contentType = $request->getHeader('content-type')[0] ?? '';
```

Comment:
`$request->getHeader()` returns a string or null, no need for the `[0]`.
```suggestion
        $contentType = $request->getHeader('content-type') ?? '';
```